agent/connect_ca_endpoint_test.go

```diff
@@ -300,10 +301,43 @@ func TestConnectCARoots_PEMEncoding(t *testing.T) {
 
 	data, err := ioutil.ReadAll(resp.Body)
 	require.NoError(t, err)
-
 	pool := x509.NewCertPool()
-
 	require.True(t, pool.AppendCertsFromPEM(data))
 	// expecting the root cert from dc1 and an intermediate in dc2
 	require.Len(t, pool.Subjects(), 2)
 }
+
+func Test_writeCA(t *testing.T) {
+	type args struct {
+		roots []*structs.CARoot
+	}
+	const ca = "-----BEGIN CERTIFICATE-----\nMIIB3TCCAYKgAwIBAgIIHk4Xdb5VAukwCgYIKoZIzj0EAwIwFDESMBAGA1UEAxMJ\nVGVzdCBDQSAxMB4XDTIxMDYxNjE1MjExOFoXDTMxMDYxNjE1MjExOFowFDESMBAG\nA1UEAxMJVGVzdCBDQSAxMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEQ268w92S\n3z1FUBjDA85hYLAFvHWEj1SyhByG5xH1i0Agbj3MQDEYEWvKpLa1IWAi6thowmg+\npAvOjKlFrdN3saOBvTCBujAOBgNVHQ8BAf8EBAMCAYYwDwYDVR0TAQH/BAUwAwEB\n/zApBgNVHQ4EIgQg1GebhqO6UZ0UDbFS4J3T4fZLu1fpOlQq+GAzjG48mJIwKwYD\nVR0jBCQwIoAg1GebhqO6UZ0UDbFS4J3T4fZLu1fpOlQq+GAzjG48mJIwPwYDVR0R\nBDgwNoY0c3BpZmZlOi8vMTExMTExMTEtMjIyMi0zMzMzLTQ0NDQtNTU1NTU1NTU1\nNTU1LmNvbnN1bDAKBggqhkjOPQQDAgNJADBGAiEAhPvUug/B5NhEFZzivsKt5Xtr\ncEyoASTqwCnKCWdNy3cCIQDgX3Stt+VlgaT4YcpLbl/GzezMVgt9F3Z6dKKHf0E9\ntA==\n-----END CERTIFICATE-----"
+	const caCarriage = ca + "\n"
```

Comment:
FYI `\r` is a carriage return and `\n` is a line feed character.

Reply:
Thanks for reviewing this @rboyer 
I get all the time confused between the 2 :)